main/src/library/Debug.flix

```diff
@@ -77,48 +76,48 @@ mod Debug {
                 } else {
                     "\"" + escape(s) + "\""
                 }
-            case y: Array[Bool, rc] =>
-                import static java.util.Arrays.toString(Array[Bool, rc]): String \ {};
-                toString(y)
-            case y: Array[Char, rc] =>
-                import static java.util.Arrays.toString(Array[Char, rc]): String \ {};
-                toString(y)
-            case y: Array[Float32, rc] =>
-                import static java.util.Arrays.toString(Array[Float32, rc]): String \ {};
-                toString(y)
-            case y: Array[Float64, rc] =>
-                import static java.util.Arrays.toString(Array[Float64, rc]): String \ {};
-                toString(y)
-            case y: Array[Int8, rc] =>
-                import static java.util.Arrays.toString(Array[Int8, rc]): String \ {};
-                toString(y)
-            case y: Array[Int16, rc] =>
-                import static java.util.Arrays.toString(Array[Int16, rc]): String \ {};
-                toString(y)
-            case y: Array[Int32, rc] =>
-                import static java.util.Arrays.toString(Array[Int32, rc]): String \ {};
-                toString(y)
-            case y: Array[Int64, rc] =>
-                import static java.util.Arrays.toString(Array[Int64, rc]): String \ {};
-                toString(y)
-            case y: Array[Unit, rc] =>
-                import static java.util.Arrays.toString(Array[##java.lang.Object, rc]): String \ {};
-                toString(unchecked_cast(y as Array[##java.lang.Object, rc]))
-            case y: Array[BigInt, rc] =>
-                import static java.util.Arrays.toString(Array[##java.lang.Object, rc]): String \ {};
-                toString(unchecked_cast(y as Array[##java.lang.Object, rc]))
-            case y: Array[BigDecimal, rc] =>
-                import static java.util.Arrays.toString(Array[##java.lang.Object, rc]): String \ {};
-                toString(unchecked_cast(y as Array[##java.lang.Object, rc]))
-            case y: Array[String, rc] =>
-                import static java.util.Arrays.toString(Array[##java.lang.Object, rc]): String \ {};
-                toString(unchecked_cast(y as Array[##java.lang.Object, rc]))
+            case y: Array[Bool, _] =>
+                import static java.util.Arrays.toString(Array[Bool, Static]): String \ {};
+                toString(unchecked_cast(y as Array[Bool, Static]))
```

Comment:
What's the inferred type of y compared to the type you cast to?

Reply:
`y` effectively has the type `exists r. Array[Bool, r]` So there's no type we can give to the import that would not require a cast.